bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,2020 @@
+        firstMotionAlarm = false;
+        motionAlarmUpdateSnapshot = false;
+        motionDetected = false;
+        if (streamingAutoFps) {
+            stopSnapshotPolling();
+        } else if (updateImageEvents.contains("4")) { // During Motion Alarms
+            stopSnapshotPolling();
+        }
+    }
+
+    // Change alarms that are not counted as motion detecting.
+    public void changeAlarmState(String thisAlarmsChannel, String state) {
+        updateState(thisAlarmsChannel, OnOffType.valueOf(state));
+    }
+
+    public void motionDetected(String thisAlarmsChannel) {
+        updateState(CHANNEL_LAST_MOTION_TYPE, new StringType(thisAlarmsChannel));
+        updateState(thisAlarmsChannel, OnOffType.valueOf("ON"));
+        motionDetected = true;
+        if (streamingAutoFps) {
+            startSnapshotPolling();
+        }
+        if (updateImageEvents.contains("2")) {
+            if (!firstMotionAlarm) {
+                if (!snapshotUri.isEmpty()) {
+                    sendHttpGET(snapshotUri);
+                }
+                firstMotionAlarm = true;// reset back to false when the jpg arrives.
+            }
+        } else if (updateImageEvents.contains("4")) { // During Motion Alarms
+            if (!snapshotPolling) {
+                startSnapshotPolling();
+            }
+            firstMotionAlarm = true;
+            motionAlarmUpdateSnapshot = true;
+        }
+    }
+
+    public void audioDetected() {
+        updateState(CHANNEL_AUDIO_ALARM, OnOffType.valueOf("ON"));
+        if (updateImageEvents.contains("3")) {
+            if (!firstAudioAlarm) {
+                if (!snapshotUri.isEmpty()) {
+                    sendHttpGET(snapshotUri);
+                }
+                firstAudioAlarm = true;// reset back to false when the jpg arrives.
+            }
+        } else if (updateImageEvents.contains("5")) {// During audio alarms
+            firstAudioAlarm = true;
+            audioAlarmUpdateSnapshot = true;
+        }
+    }
+
+    public void noAudioDetected() {
+        setChannelState(CHANNEL_AUDIO_ALARM, OnOffType.valueOf("OFF"));
+        firstAudioAlarm = false;
+        audioAlarmUpdateSnapshot = false;
+    }
+
+    public String returnValueFromString(String rawString, String searchedString) {
+        String result = "";
+        int index = rawString.indexOf(searchedString);
+        if (index != -1) // -1 means "not found"
+        {
+            result = rawString.substring(index + searchedString.length(), rawString.length());
+            index = result.indexOf("\r\n"); // find a carriage return to find the end of the value.
+            if (index == -1) {
+                return result; // Did not find a carriage return.
+            } else {
+                return result.substring(0, index);
+            }
+        }
+        return ""; // Did not find the String we were searching for
+    }
+
+    public String searchString(String rawString, String searchedString) {
```

Comment:
Can you add comments or javadoc for this method so we know what this method should be doing. It is difficult to tell if this code below is correct if you don't know what it should be doing.

Reply:
DONE: Added javdoc and moved the function to the only class that used it and made it private.